ProxySQL 2.0.18 gr_member_routing_candidate_status.viable_candidate return NULL causes infinite crash reboot

## Version info

we use ProxySQL2.0.18.
```
mysql> select version();
+-------------------+
| version()         |
+-------------------+
| 2.0.18-7-gc45810f |
+-------------------+
1 row in set (0.00 sec)
```
Percona MySQL 8.0.22.
```
root@localhost 10:03:55 [(none)]> select version();
+-----------+
| version() |
+-----------+
| 8.0.22-13 |
+-----------+
1 row in set (0.00 sec)
```

OS version is CentOS Linux release 7.4.1708 (Core) 

Single Primary Group Replication.

## How to reproduce:
1. follow the [proxysql offical doc](https://proxysql.com/documentation/main-runtime/#mysql_group_replication_hostgroups)
  config mysql group replication need to add view to mysql instance sys
![image](https://user-images.githubusercontent.com/24428128/115101759-65a36b80-9f79-11eb-973e-2b236a303928.png)

2. create a three node singe primary mysql group replication cluster
```
root@localhost 12:27:16 [sys]> select * from performance_schema.replication_group_members;
+---------------------------+--------------------------------------+---------------+-------------+--------------+-------------+----------------+
| CHANNEL_NAME              | MEMBER_ID                            | MEMBER_HOST   | MEMBER_PORT | MEMBER_STATE | MEMBER_ROLE | MEMBER_VERSION |
+---------------------------+--------------------------------------+---------------+-------------+--------------+-------------+----------------+
| group_replication_applier | 112f59cb-9f27-11eb-834f-0050563b7b42 | 172.16.120.10 |        3306 | ONLINE       | PRIMARY     | 8.0.22         |
| group_replication_applier | 1349b4d5-9f27-11eb-8213-0050562f2198 | 172.16.120.11 |        3306 | ONLINE       | SECONDARY   | 8.0.22         |
| group_replication_applier | 136e6aa7-9f27-11eb-bfa3-0050563108d2 | 172.16.120.12 |        3306 | ONLINE       | SECONDARY   | 8.0.22         |
+---------------------------+--------------------------------------+---------------+-------------+--------------+-------------+----------------+
```
3. shutdown one mysql node, let's shutdown 172.16.120.12
4. start 172.16.120.12
5. query sys.gr_member_routing_candidate_status
```
root@localhost 12:07:10 [sys]> select * from sys.gr_member_routing_candidate_status;
+------------------+-----------+---------------------+----------------------+
| viable_candidate | read_only | transactions_behind | transactions_to_cert |
+------------------+-----------+---------------------+----------------------+
| NULL             | YES       |                   0 |                 NULL |
+------------------+-----------+---------------------+----------------------+
1 row in set (0.00 sec)
```
**you can see viable_candidate is `NULL`**

6. check proxysql
```
proxysql(_Z13crash_handleri+0x1a)[0x4b86da]
2021-04-17 12:10:48 main.cpp:1564:ProxySQL_daemonize_phase3(): [ERROR] ProxySQL crashed. Restarting!
proxysql(_Z13crash_handleri+0x1a)[0x4b86da]
2021-04-17 12:10:48 main.cpp:1564:ProxySQL_daemonize_phase3(): [ERROR] ProxySQL crashed. Restarting!
proxysql(_Z13crash_handleri+0x1a)[0x4b86da]
2021-04-17 12:10:49 main.cpp:1564:ProxySQL_daemonize_phase3(): [ERROR] ProxySQL crashed. Restarting!
proxysql(_Z13crash_handleri+0x1a)[0x4b86da]
2021-04-17 12:10:51 main.cpp:1564:ProxySQL_daemonize_phase3(): [ERROR] ProxySQL crashed. Restarting!
proxysql(_Z13crash_handleri+0x1a)[0x4b86da]
2021-04-17 12:10:51 main.cpp:1564:ProxySQL_daemonize_phase3(): [ERROR] ProxySQL crashed. Restarting!
proxysql(_Z13crash_handleri+0x1a)[0x4b86da]
2021-04-17 12:10:52 main.cpp:1564:ProxySQL_daemonize_phase3(): [ERROR] ProxySQL crashed. Restarting!



Error: signal 11:
/usr/bin/proxysql(_Z13crash_handleri+0x1a)[0x4b86da]
/lib64/libc.so.6(+0x36280)[0x7f8c4cf6c280]
/lib64/libc.so.6(+0x1463f1)[0x7f8c4d07c3f1]
/usr/bin/proxysql(_Z32monitor_group_replication_threadPv+0xe99)[0x5861a9]
/usr/bin/proxysql(_ZN14ConsumerThread3runEv+0xf7)[0x589b27]
/lib64/libpthread.so.0(+0x7dd5)[0x7f8c4e14bdd5]
/lib64/libc.so.6(clone+0x6d)[0x7f8c4d033ead]
2021-04-16 17:44:49 main.cpp:1564:ProxySQL_daemonize_phase3(): [ERROR] ProxySQL crashed. Restarting!
```
**infinite crash rebooting!**


## How to solve this problem?
use IFNULL
```
CREATE VIEW gr_member_routing_candidate_status AS
    SELECT 
        IFNULL(sys.gr_member_in_primary_partition(),'NO') AS viable_candidate,
        IF((SELECT 
                    ((SELECT 
                                GROUP_CONCAT(performance_schema.global_variables.VARIABLE_VALUE
                                        SEPARATOR ',')
                            FROM
                                performance_schema.global_variables
                            WHERE
                                (performance_schema.global_variables.VARIABLE_NAME IN ('read_only' , 'super_read_only'))) <> 'OFF,OFF')
                ),
            'YES',
            'NO') AS read_only,
        IFNULL(sys.gr_applier_queue_length(),0) AS transactions_behind,
        IFNULL(sys.gr_transactions_to_cert(),0) AS transactions_to_cert;
```
Then repeat the above steps, proxysql no longer crashes

😓😓😓😓😓😓😓😓😓😓😓😓😓😓

--- a/lib/MySQL_Monitor.cpp
+++ b/lib/MySQL_Monitor.cpp
@@ -1501,13 +1501,15 @@ __exit_monitor_group_replication_thread:
 				goto __end_process_group_replication_result2;
 			}
 			MYSQL_ROW row=mysql_fetch_row(mmsd->result);
-			if (!strcasecmp(row[0],"YES")) {
+			if (row[0] && !strcasecmp(row[0],"YES")) {
 				viable_candidate=true;
 			}
-			if (!strcasecmp(row[1],"NO")) {
+			if (row[1] && !strcasecmp(row[1],"NO")) {
 				read_only=false;
 			}
-			transactions_behind=atol(row[2]);
+			if (row[2]) {
+				transactions_behind=atol(row[2]);
+			}
 			mysql_free_result(mmsd->result);
 			mmsd->result=NULL;
 		}

